internal/cmd/protoc-gen-go-flipt-sdk/main.go

```diff
@@ -234,27 +234,36 @@ type ClientTokenProvider interface {
 // lifecycle support.
 type SDK struct {
 	transport Transport
-    tokenProvider ClientTokenProvider
+    authenticationProvider ClientAuthenticationProvider
 }
 
 // Option is a functional option which configures the Flipt SDK.
 type Option func(*SDK)
 
-// WithClientTokenProviders returns an Option which configures
-// any supplied SDK with the provided ClientTokenProvider.
-func WithClientTokenProvider(p ClientTokenProvider) Option {
+// WithAuthenticationProviders returns an Option which configures
+// any supplied SDK with the provided ClientAuthenticationProvider.
+func WithAuthenticationProvider(p ClientAuthenticationProvider) Option {
 	return func(s *SDK) {
-        s.tokenProvider = p
+        s.authenticationProvider = p
```

Comment:
Seems like a breaking change? Do we want to break this?

We could instead just add a `WithJWTTokenProvider` method alongside the existing `WithClientTokenProvider`?
Or we break it, up to you.

Reply:
true. can redo and make this a non-breaking change

Reply:
actually looking more into this, what would the arguments of `WithJWTTokenProvider` be? If a separate type, ie a `JWTTokenProvider` then what would that type be? Like how would it be different than `ClientTokenProvider`?

Reply:
I see what you're saying. Then perhaps just leave `WithClientTokenProvider` and mark it deprecated?
Then create a WithAuthenticationProvider which just delegates to it for now? wdyt?